x/auth/ante.go

```diff
@@ -277,24 +277,36 @@ func deductFees(acc Account, fee StdFee) (Account, sdk.Result) {
 	coins := acc.GetCoins()
 	feeAmount := fee.Amount
 
-	newCoins := coins.Minus(feeAmount)
-	if !newCoins.IsNotNegative() {
+	if !feeAmount.IsValid() {
+		return nil, sdk.ErrInsufficientFee(fmt.Sprintf("invalid fee amount: %s", feeAmount)).Result()
+	}
+
+	newCoins, ok := coins.SafeMinus(feeAmount)
```

Comment:
why are we not using plain `Minus` ?

Reply:
Because `Minus` will panic on any negative amount! So here want to gracefully return the error. As mentioned in a similar conversation above, I think we can do `IsAllLTE` here instead and then once sure, we can safely do subtraction.